core/trino-main/src/test/java/io/trino/execution/TestCreateTableTask.java

```diff
@@ -349,6 +357,38 @@ public void testUnsupportedCreateTableWithField()
                 .hasMessage("Column name 'a.b' must not be qualified");
     }
 
+    @Test
+    public void testCreateTableWithCoercedType()
+    {
+        CreateTable statement = new CreateTable(QualifiedName.of("test_table"),
+                ImmutableList.of(new ColumnDefinition(QualifiedName.of("a"), toSqlType(TIMESTAMP_NANOS), true, emptyList(), Optional.empty())),
+                IGNORE,
+                ImmutableList.of(),
+                Optional.empty());
+        CreateTableTask createTableTask = new CreateTableTask(plannerContext, new AllowAllAccessControl(), columnPropertyManager, tablePropertyManager);
+        getFutureValue(createTableTask.internalExecute(statement, testSession, List.of(), output -> {}));
+        assertThat(metadata.getReceivedTableMetadata().get(0).getColumns().get(0).getType()).isEqualTo(TIMESTAMP_MILLIS);
+    }
+
+    @Test
+    public void testCreateTableLikeWithCoercedType()
+    {
+        CreateTable statement = new CreateTable(
```

Comment:
nit 

        CreateTable statement = new CreateTable(
                QualifiedName.of("test_table"),
                ImmutableList.of(
                        new LikeClause(
                                QualifiedName.of(PARENT_TABLE_WITH_COERCED_TYPE.getTable().getTableName()), 
                                Optional.of(INCLUDING))),
                IGNORE,
                ImmutableList.of(),
                Optional.empty());